modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -4437,11 +4447,16 @@ private void _startWorkflowInstance(
 			_objectDefinitionPersistence.findByPrimaryKey(
 				objectEntry.getObjectDefinitionId());
 
+		boolean skipObjectActionExecution =
+			ObjectActionThreadLocal.isSkipObjectActionExecution();
+
 		boolean workflowEnabled = WorkflowThreadLocal.isEnabled();
 
 		try {
 			_skipModelListeners.set(skipModelListener);
 
+			ObjectActionThreadLocal.setSkipObjectActionExecution(true);
```

Comment:
Why do you need to change the value of `skipObjectActionExecution` property here in activate method?

Reply:
The `_startWorkflowInstance` method is called in the `addObjectEntry` and the `updateObjectEntry` methods. And it will eventually call the `updateStatus` method. 

In this case, we do not this method to trigger any actions, because we already passed through the add/update methods. We only want it to trigger actions when a workflow task is completed.

Reply:
Sorry, I made a confusion thinking it was being called in the activate method, but it is being called in the `_startWorkflowInstance` method. Looks fine to me. Thanks!